x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph/fps_trendline.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import { CommonProps } from '@elastic/eui';
+
+export const FpsTrendline: React.FC<CommonProps> = (props: CommonProps) => {
+  const [fpsSamples, setFpsSamples] = useState<number[]>([]);
+  const frameCount = useRef(0);
+  const lastTimestamp = useRef(performance.now());
+
+  useEffect(() => {
+    let animationFrameId: number;
+
+    const calculateFPS = (timestamp: number) => {
+      frameCount.current += 1;
+      const delta = timestamp - lastTimestamp.current;
+
+      if (delta >= 1000) {
+        const fps = (frameCount.current * 1000) / delta;
+        setFpsSamples((prevSamples) => {
+          const updatedSamples = [...prevSamples, fps];
+          return updatedSamples.slice(-20);
+        });
+        frameCount.current = 0;
+        lastTimestamp.current = timestamp;
+      }
+
+      animationFrameId = requestAnimationFrame(calculateFPS);
+    };
+
+    animationFrameId = requestAnimationFrame(calculateFPS);
+
+    return () => cancelAnimationFrame(animationFrameId);
+  }, []);
```

Comment:
maybe we can abstract this hook and call it `useFPSCounter` or similar that runs any custom logic inside. It's very similar to the hook you wrote in `fps_monitor.tsx`

Reply:
sorry, that was a duplicated code. I deleted it